tiled/_tests/test_awkward.py

```diff
@@ -116,3 +116,14 @@ def test_export_parquet(tmpdir):
         actual = pyarrow.parquet.read_table(filepath)
         expected = awkward.to_arrow_table(array)
         assert actual == expected
+
+
+def test_long_url(tmpdir):
+    # https://github.com/bluesky/tiled/pull/577
+    catalog = in_memory(writable_storage=tmpdir)
+    app = build_app(catalog)
+    with Context.from_app(app) as context:
+        client = from_context(context)
+        array = awkward.Array([{f"key{i:05}": i for i in range(4000)}])
+        aac = client.write_awkward(array, key="test")
+        aac[0]
```

Comment:
This test induces the error in the main branch and passes with the accompanying codebase changes in the PR.  However, the reason for failing or passing the test is abstruse--it's obscured by what's happening under the hood in `AwkwardClient.read()`.

I have some ideas how to clarify this, but I've "timed out" for the day.  I'd like to push a suggestion tomorrow.